src/errors.ts

```diff
@@ -1,6 +1,50 @@
+import { t } from '@lingui/macro'
+
 export class IntegrationError extends Error {
   constructor(message: string) {
     super(message)
     this.name = 'Integration Error'
   }
 }
+
+const DefaultErrorHeader = t`Please refresh the page and try again.`
+const DefaultErrorAction = t`Reload the page`
+
+export abstract class WidgetError extends Error {
+  header: string = DefaultErrorHeader
```

Comment:
nit: `DefaultErrorHeader` is a string, so no explicit typing is needed. Same with `action`.